flink/src/main/java/org/apache/iceberg/flink/source/enumerator/ContinuousSplitPlanner.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.flink.source.enumerator;
+
+public interface ContinuousSplitPlanner {
+
+  ContinuousEnumerationResult planSplits(IcebergEnumeratorPosition lastPosition);
```

Comment:
Instead of passing in the position, do you want to pass the EnumeratorState instead? I'm asking this because there might be cases where the planner might decide to enumerate splits only when the previously returned splits have all been completely read. So, passing in the EnumeratorState will help the planner identify where the source is to decide if it wants to do the enumeration further.